src/algebra/lie/basic.lean

```diff
@@ -570,21 +570,36 @@ instance lie_subalgebra_lie_ring (L' : lie_subalgebra R L) : lie_ring L' :=
 instance lie_subalgebra_lie_algebra (L' : lie_subalgebra R L) : lie_algebra R L' :=
 { lie_smul := by { intros, apply set_coe.ext, apply lie_smul } }
 
-@[simp] lemma lie_subalgebra.mem_coe {L' : lie_subalgebra R L} {x : L} :
+namespace lie_subalgebra
+
+@[simp] lemma mem_coe {L' : lie_subalgebra R L} {x : L} :
   x ∈ (L' : set L) ↔ x ∈ L' := iff.rfl
 
-@[simp] lemma lie_subalgebra.mem_coe' {L' : lie_subalgebra R L} {x : L} :
+@[simp] lemma mem_coe' {L' : lie_subalgebra R L} {x : L} :
   x ∈ (L' : submodule R L) ↔ x ∈ L' := iff.rfl
 
-@[simp, norm_cast] lemma lie_subalgebra.coe_bracket (L' : lie_subalgebra R L) (x y : L') :
+@[simp, norm_cast] lemma coe_bracket (L' : lie_subalgebra R L) (x y : L') :
   (↑⁅x, y⁆ : L) = ⁅(↑x : L), ↑y⁆ := rfl
 
-@[ext] lemma lie_subalgebra.ext (L₁' L₂' : lie_subalgebra R L) (h : ∀ x, x ∈ L₁' ↔ x ∈ L₂') :
+@[ext] lemma ext (L₁' L₂' : lie_subalgebra R L) (h : ∀ x, x ∈ L₁' ↔ x ∈ L₂') :
   L₁' = L₂' :=
 by { cases L₁', cases L₂', simp only [], ext x, exact h x, }
 
-lemma lie_subalgebra.ext_iff (L₁' L₂' : lie_subalgebra R L) : L₁' = L₂' ↔ ∀ x, x ∈ L₁' ↔ x ∈ L₂' :=
-⟨λ h x, by rw h, lie_subalgebra.ext R L L₁' L₂'⟩
+lemma ext_iff (L₁' L₂' : lie_subalgebra R L) : L₁' = L₂' ↔ ∀ x, x ∈ L₁' ↔ x ∈ L₂' :=
+⟨λ h x, by rw h, ext R L L₁' L₂'⟩
+
+@[simp] lemma coe_to_set_mk (S : set L) (h₁ h₂ h₃ h₄) :
+  ((⟨S, h₁, h₂, h₃, h₄⟩ : lie_subalgebra R L) : set L) = S := rfl
+
+@[simp, norm_cast] theorem coe_set_eq_iff (L₁' L₂' : lie_subalgebra R L) :
+  (L₁' : set L) = L₂' ↔ L₁' = L₂' :=
+by { cases L₁'; cases L₂'; simp, }
```

Comment:
Over in `submodule` this is written as

https://github.com/leanprover-community/mathlib/blob/f3d3d0461cbf550eed1ac770e48f6c2c51bb9c01/src/algebra/module/submodule.lean#L62-L65

It would be nice to have similar lemma names here